instruqt/protecting-your-assets/06-validating-license-fields/assignment.md

```diff
@@ -0,0 +1,188 @@
+    "title": "Member Count",
+    "value": 100,
+    "valueType": "Integer",
+    "signature": {
+      "v1": "d4oAFJ8trT1SVwnDa5xkURT25TD3gQwaS/urGMaAC4vOx+nKItEgI8jLySyM4ST6X+7MkJ4r4Sb9qpL9+JzeSkZA+pNGqZonebg7MaLHme3sAmw7v4+TPn+MO53fdEnVPhsZj33m6EUQvhORhxOQlVqwjVjlTbHBWidmnUaQkUr7RXLPhVL6RQJxwde6yaObifxY0Ux2BL6nbxTIwMh/yBchx8GR+gHyi6MdEBbT80P3AxzOAVPflpK2y2eX84b8wST9zHel4dL+FTPftXdqVbJB/99TIFmeiKONCi7JMq+6rGiXqgOgPg0h5YVgiS1UlLML47IrBT74Rq+EGnAE6g=="
+    }
+  }
+}
+```
+
+To get a single field, add the field name to the URI.
+
+```
+curl http://replicated:3000/api/v1/license/fields/member_count_max | jq .
+```
+
+This will give you the value for the "Member Count" entitlement you added to
+the license earlier.
+
+```
+{
+  "name": "member_count_max",
+  "title": "Max Member Count",
+  "value": 100,
+  "valueType": "Integer",
+  "signature": {
+    "v1": "ttq5sqIQbY3U+LgbqEKMQuKn5UO5r3tvHnc18Cp1JPEUGX7f6X7ZjkKO4UkTHqlyC3ZFxVc0e6FVDH02+Qw4b9biPShFIPz0wzoWdGzzuNGWVfGUIQKXYR3WXLapKaaOeeJLM6c/Z90MzW9cwxBrUVbyAzaBagBPoegL0/PpKhM2NUKpDYeDHPRHnFgANXDE+5IoB3EG4lSBivkWnioAmWZIbVcHHZTZbE9Bwdh6Tmo/J3NUFcSBUjaS23vk5y8pVQ9otefxt5DVUvZb3r6RST/xzAOLKvk0tZExRb2X/CzdmiFJct3g9sJXPMzQe/RBPUUQFtbJa+K1aHKgyK4AmA=="
+  }
+}
+```
+
+When you call the SDK from your code you can access the value in the `value`
+field. Here in the shell we just change our `jq` expression a little bit.
+
+```
+curl http://replicated:3000/api/v1/license/fields/member_count_max | jq .value
+```
+
+Disconnect from the client pod for now. We'll use it again when we validate the
+signed value.
+
+Verifying License Signatures
+============================
+
+You may not want to trust that value without verifying it. It's not likely your
+customer is trying to circumvent your license, but it's possible that value was
+changed whether out of curiosity or malice. That's why the license and it's
+fields are signed.
+
+The "Settings" page for the application in the Vendor Portal shows your public
+key and provides example code in a few programming languages. The key is also
+available from the Vendor Portal API. Let's take a quick look. If you're not
+still logged in use the following username and password to log back in.
+
+Username: `[[ Instruqt-Var key="USERNAME" hostname="shell" ]]`<br/>
+Password: `[[ Instruqt-Var key="PASSWORD" hostname="shell" ]]`
+
+Navigate to "Settings" in the left menu, then select the "Replicated SDK
+signature verification" tab.
+
+![Replicated SDK Signature Verification Instructions on the Vendor Portal](../assets/signature-verification-info.png)
+
+To save you some time, the lab setup grabbed the public key for your
+application and save it in the file `signing-key.pub`. We're going to use this
+key along with the information from the SDK to verify that the member count has
+not been tampered with. Copy the public key file to the `sdk-client` pod we
+created earlier.
+
+```
+kubectl -n slackernews cp signing-key.pub sdk-client:signing-key.pub
+```
```

Comment:
```
replicant@shell:~$ kubectl -n slackernews cp signing-key.pub sdk-client:signing-key.pub
error: signing-key.pub doesn't exist in local filesystem
```

It's not there for me?

Unfortunately it timed out on me again at this point. I'm going to wait until there's a longer timeout before trying to get further.

Reply:
The setup code should be putting the key there. Likely I'm putting it in the wrong directory or not changing the perms.

Reply:
File was in the wrong directory. Should be getting saved to the right place now.

Reply:
I think I figured out what I did with the timeouts and they should be adjusted.